### PR TITLE
Move google auth to application controller

### PR DESCRIPTION
**Pull request by stephaniegiang**
## What
Part 1 of https://github.com/education/classroom/issues/2027

Moves Google authentication into `ApplicationController`

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -45,8 +45,30 @@ def failbot_context
     }
   end
 
+  def authorize_google_classroom
```
Comment (stephaniegiang) [inline app/controllers/application_controller.rb]: with `app/controllers/application_controller/google_authentication_dependency.rb` not working, I just added the methods to the actual controller. This seemed to fix the issue I was having with the file.

File: app/controllers/application_controller/google_authentication_dependency.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class ApplicationController
```
Comment (stephaniegiang) [inline app/controllers/application_controller/google_authentication_dependency.rb]: So I tried making a new dependency to add it to the `ApplicationController` but this didn't really work. The methods weren't added for some reason. Am I missing something?

Comment (thebradbain) [reply app/controllers/application_controller/google_authentication_dependency.rb]: Ah! I ran into this issue too -- 

you need to add your file to the `depends_on:` line within `ApplicationController`

Comment (stephaniegiang) [reply app/controllers/application_controller/google_authentication_dependency.rb]: Oh really? I knew I was missing something 😂

**Comment by stephaniegiang** [comment]
@d12 this PR is ready!

**Pull request merged**